_1327/polls/templates/polls_edit.html

```diff
@@ -4,74 +4,74 @@
+					</tr>
+				{% endfor %}
+			</tbody>
+		</table>
+	</div>
 {% endblock %}
 
 {% block additional_scripts%}
-    // enable formset functionality
-    $('.choice-row').formset({
-        prefix: '{{ formset.prefix }}',
-        deleteCssClass: 'delete-button btn btn-default',
-        deleteText: '{% trans "Delete" %}',
-        added: function(row) {
-            row.find('.input-group-btn').append(row.find('.delete-button'));
-            var form = $('#choice-inline-form');
-            form.append(row);
-        }
-    });
-    $('.delete-button').each(function() {
-        $(this).prev(".form-group.inlineformset").find(".input-group-btn").append(this);
-    });
-    $('.add-row-container').append($('.add-row'));
-    $('.add-row').addClass('btn btn-default');
-    $('.add-row').text('{% trans "Add another choice" %}');
+	function setOrderIndex(row, index) {
+		var text = row.find("input[name*='text']");
+		var description = row.find("input[name*='description']");
+		var orderIndex = row.children("input[name*='index']");
+		if (text.val() === "" && description.val() === "") {
+			// if the row is empty (has no text in the input fields) set the order to 0 (default),
+			// so that the one extra row doesn't change its initial value
+			orderIndex.val(0);
+		} else {
+			orderIndex.val(index);
+		}
+	}
 
-    $('#document-form').on('submit', function(event) {
-        // go through all choices and set the ordering index according to the current position
-        $('#choice-inline-form').children().each(function (index, element) {
-            element = $(element);
-            if (element.find('.choice-text').attr('value') == undefined && element.find('.choice-description').attr('value') == undefined) {
-                // empty element that should not have any new data
-                element.find('.ordering-index').attr('value', 0);
-                return
-            }
-            $(element).attr('value', index);
-        });
-    });
+	$('#document-form').on('submit', function(event) {
+		// go through all choices and set the ordering index according to the current position
+		var rows = $('.sortable');
+		rows.each(function (index, row) {
+			setOrderIndex($(row), index)
+		});
+		return true;
+	});
 
-    // add datepicker to date inputs
-    $("input[name*='date']:not([readonly='True'])").datepicker( $.datepicker.regional["{{ LANGUAGE_CODE }}"] );
+	// enable formset functionality
+	$('#choice-inline-form table tbody tr').formset({
+		prefix: '{{ formset.prefix }}',
+		deleteCssClass: 'delete',
+		added: function(row) {
+			row.find("input[name*='index']").val(row.parent().children().length);
+		}
+	});
+
+	// make forms sortable
+	$('#choice-inline-form table tbody').sortable({
+		items: '.sortable',
+		handle: '.sorthandle',
+	});
+
+	// add datepicker to date inputs
+	$("input[name*='date']:not([readonly='True'])").datepicker( $.datepicker.regional["{{ LANGUAGE_CODE }}"] );
```

Comment:
its actually necessary right now. We would not have a datepicker without this line ;) but this is something we could put into to master template for editing documents